component/block_cache/block_cache.go

```diff
@@ -1788,9 +1797,26 @@ func (bc *BlockCache) diskEvict(node *list.Element) {
 	_ = os.Remove(localPath)
 }
 
+// getDiskUsage returns the current disk usage in MB
+func (bc *BlockCache) getDiskUsage() (float64, error) {
```

Comment:
This function looks ok but if we call this function periodically or at high interval then we are comparing the flags everytime. This can be optimised further by creating a function pointer to selected method and in this function just call whatever its poitning to,